src/filter/http_context.rs

```diff
@@ -228,7 +228,36 @@ impl Filter {
                     }
                 }
                 RLA_action_specifier::dynamic_metadata(_) => todo!(),
-                RLA_action_specifier::metadata(_) => todo!(),
+                RLA_action_specifier::metadata(md) => {
+                    // Note(rahul): defaulting to dynamic metadata source right now.
+                    let metadata_key = &md.get_metadata_key().key;
+                    let mut metadata_path: Vec<&str> = md
+                        .get_metadata_key()
+                        .get_path()
+                        .iter()
+                        .map(|path_segment| path_segment.get_key())
+                        .collect();
+                    let default_value = md.get_default_value();
```

Comment:
default value is a required fields? in the envoy docs, it is not https://www.envoyproxy.io/docs/envoy/latest/api-v3/config/route/v3/route_components.proto#envoy-v3-api-msg-config-route-v3-ratelimit-action-metadata

Maybe it is in the protoc objects.

What if the default value is not set?

```
default_value

    ([string](https://developers.google.com/protocol-buffers/docs/proto#scalar)) An optional value to use if metadata_key is empty. If not set and no value is present under the metadata_key then no descriptor is generated.
```

Reply:
It would use the default `String` value i.e. `""`

Reply:
Looks like it will fail the parsing of configuration and yes, it's due to protoc objects.

Reply:
I wonder if `default_value` is empty then we should skip this action altogether.

Reply:
I would say so

Reply:
I think a better option is to let RLP validation tell the user that the default value is not provided.

Reply:
Done.